File: hsg9-tinh/2023-2024-haiduong/songuyento/songuyento.cpp
#include <iostream>

#define input_file "songuyento.inp"
#define output_file "songuyento.out"

using namespace std;

int n;
int result;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> n;
}


bool is_prime(int x)
{
    if (x < 1) return false;

    if (x < 4) return true;

    if (x % 2 == 0) return false;

    if (x % 3 == 0) return false;

    for (int i = 5; i * i < x + 1; i += 6)
    {
        if (x % i == 0) return false;
        if (x % (i + 2) == 0) return false;
    }

    return true;
}


void process()
{
    while (true)
    {
        if (is_prime(n))
        {
            result = n;
            break;
        }

        n++;
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << result;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-2024-haiduong/songuyento/songuyento.py
import os
import sys
import math

input_file = os.path.join(sys.path[0], 'songuyento.inp')
output_file = os.path.join(sys.path[0], 'songuyento.out')

n = result = 0


def input():
    global n

    with open(input_file, 'r') as f:
        n = int(f.readline())


def is_prime(x):
    if x < 2:
        return False
    
    if x < 4:
        return True

    if x % 2 == 0 or x % 3 == 0:
        return False
    
    for i in range(5, int(math.sqrt(x)) + 1, 6):
        if x % i == 0 or x % (i + 2) == 0:
            return False
        
    return True


def process():
    global n, result

    while True:
        if is_prime(n):
            result = n
            break
    
        n += 1


def output():
    global result

    with open(output_file, 'w') as f:
        f.write(str(result))


if __name__ == '__main__':
    input()
    process()
    output()